updated boost m4

toanju opened the pull request:
current boost m4 is not working any more on latest fedora

Commit:
updated boost m4

current boost m4 is not working any more on latest fedora

msune commented:
Thanks. This PR however doesn't seem to get along correctly with older versions:

```
marc@dev:~/git/bisdn/xdpd$ bash autogen.sh 
Populating git submodules...
Done
Running autoreconf...
config/plugin_rest.m4:9: warning: AC_REQUIRE: `BOOST_REQUIRE' was expanded before it was required
config/plugin_rest.m4:9: http://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required
config/boost.m4:311: BOOST_FIND_LIBS is expanded from...
config/boost.m4:354: BOOST_FIND_LIB is expanded from...
config/boost.m4:1027: BOOST_SYSTEM is expanded from...
config/boost.m4:546: BOOST_ASIO is expanded from...
/usr/share/aclocal-1.14/cond-if.m4:23: AM_COND_IF is expanded from...
config/plugins.m4:5: ADD_PLUGIN is expanded from...
config/plugin_rest.m4:9: the top level
config/plugin_rest.m4:9: warning: AC_REQUIRE: `BOOST_REQUIRE' was expanded before it was required
config/plugin_rest.m4:9: http://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required
config/boost.m4:311: BOOST_FIND_LIBS is expanded from...
config/boost.m4:354: BOOST_FIND_LIB is expanded from...
config/boost.m4:1027: BOOST_SYSTEM is expanded from...
config/boost.m4:546: BOOST_ASIO is expanded from...
/usr/share/aclocal-1.14/cond-if.m4:23: AM_COND_IF is expanded from...
config/plugins.m4:5: ADD_PLUGIN is expanded from...
config/plugin_rest.m4:9: the top level
config/plugin_rest.m4:9: warning: AC_REQUIRE: `BOOST_REQUIRE' was expanded before it was required
config/plugin_rest.m4:9: http://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required
config/boost.m4:311: BOOST_FIND_LIBS is expanded from...
config/boost.m4:354: BOOST_FIND_LIB is expanded from...
config/boost.m4:1027: BOOST_SYSTEM is expanded from...
config/boost.m4:546: BOOST_ASIO is expanded from...
aclocal.m4:408: AM_COND_IF is expanded from...
config/plugins.m4:5: ADD_PLUGIN is expanded from...
config/plugin_rest.m4:9: the top level
```

Enviornment:

```
marc@dev:~/git/bisdn/xdpd$ uname -a
Linux dev 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt9-3~deb8u1 (2015-04-24) x86_64 GNU/Linux
marc@dev:~/git/bisdn/xdpd$ autoconf --version
autoconf (GNU Autoconf) 2.69
Copyright (C) 2012 Free Software Foundation, Inc.
License GPLv3+/Autoconf: GNU GPL version 3 or later
<http://gnu.org/licenses/gpl.html>, <http://gnu.org/licenses/exceptions.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David J. MacKenzie and Akim Demaille.
marc@dev:~/git/bisdn/xdpd$ automake --version
automake (GNU automake) 1.14.1
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl-2.0.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Tom Tromey <tromey@redhat.com>
       and Alexandre Duret-Lutz <adl@gnu.org>.
```

Commit:
Use AC_REQUIRE for BOOST_REQUIRE

This patch fixes the warning:

config/plugin_rest.m4:9: warning: AC_REQUIRE: `BOOST_REQUIRE' was
expanded before it was required, part of the PR #79

msune commented:
Once https://github.com/msune/xdpd/commit/17d0cedf3631ddf7a4f1440030be4e7e816784f0 is merged, I am fine with it => so ACK.

toanju committed:
Merge pull request #1 from msune/fix_boost_require_warning

Use AC_REQUIRE for BOOST_REQUIRE
